c3d/c3d.py

```diff
@@ -1661,26 +1958,377 @@ def __init__(self,
         '''Set metadata for this writer.
 
         '''
-        # Always write INTEL format
-        self._dtypes = DataTypes(PROCESSOR_INTEL)
+        self._dtypes = DataTypes(PROCESSOR_INTEL) # Always write INTEL format
         super(Writer, self).__init__()
-        self._point_rate = point_rate
-        self._analog_rate = analog_rate
-        self._analog_per_frame = analog_rate / point_rate
-        self._point_scale = point_scale
-        self._point_units = point_units
-        self._gen_scale = gen_scale
         self._frames = []
 
-    def add_frames(self, frames):
+        # Header properties
+        self._header.frame_rate = np.float32(point_rate)
+        self._header.scale_factor = np.float32(point_scale)
+        self.analog_rate = analog_rate        
+
+        # Custom properties
+        self._point_units = point_units
+        self.set_analog_general_scale(gen_scale)
+
+    @staticmethod
+    def from_reader(reader, conversion=None):
+        '''
+        source : 'class' Manager
```

Comment:
The docstring doe snot seemed to be formatted correctly. Also source -> reader?

Reply:
Copied docstring from future commit, class names are named based on future package structure however.